src/super_gradients/training/models/detection_models/yolo_base.py

```diff
@@ -408,6 +416,36 @@ def __init__(self, backbone: Type[nn.Module], arch_params: HpmStruct, initialize
             self._head = YoloHead(self.arch_params)
             self._initialize_module()
 
+        self._class_names: Optional[List[str]] = None
+        self._image_processor: Optional[Processing] = None
+
+    @staticmethod
+    def get_post_prediction_callback(conf: float, iou: float) -> DetectionPostPredictionCallback:
+        return YoloPostPredictionCallback(conf=conf, iou=iou)
+
+    def set_dataset_processing_params(self, class_names: Optional[List[str]], image_processor: Optional[Processing]) -> None:
+        """Set the processing parameters for the dataset.
+
+        :param class_names:     (Optional) Names of the dataset the model was trained on.
+        :param image_processor: (Optional) Image processing objects to reproduce the dataset preprocessing used for training.
+        """
+        self._class_names = class_names or self._class_names
+        self._image_processor = image_processor or self._image_processor
```

Comment:
@shaydeci Also, not a big fan of having this duplicate (this is implemented in both yolobase and ppyoloe).
But they don't share a parent besides SgModule, so unless we implement this for SgModule or if we implement such an abstract  (which you and Ofri seemed to not want for now) we have to duplicate the code